Source/WebKit/UIProcess/XR/openxr/PlatformXROpenXR.cpp

```diff
@@ -253,9 +258,9 @@ void OpenXRCoordinator::endSessionIfExists(std::optional<WebCore::PageIdentifier
                 sessionEventClient->sessionDidEnd(m_deviceIdentifier);
             }
 
+            page.uiClient().didEndXRSession(page);
```

Comment:
I see you moved this before we clear `m_state`. Was this so that if the client were to query something back,`m_state` would still have the previous information? Do we allow that kind of re-entrant behaviour?